### PR TITLE
Automation-time types file

### PR DESCRIPTION
**Pull request by rhuttman**

### CODE REVIEW

**Comment by rhuttman** [comment]
I have tested this as a runtime upgrade to ensure I was able to read the chain state.

File: pallets/automation-time/src/lib.rs [hunk 1]
```diff
@@ -79,188 +81,16 @@ pub mod pallet {
...
-	}
+	pub type AccountTaskId<T> = (AccountOf<T>, TaskId<T>);
+	pub type ActionOf<T> = Action<AccountOf<T>, BalanceOf<T>, <T as Config>::CurrencyId>;
+	pub type TaskOf<T> = Task<
+		AccountOf<T>,
+		BalanceOf<T>,
+		<T as Config>::CurrencyId,
+		<T as Config>::MaxExecutionTimes,
+	>;
+	pub type MissedTaskV2Of<T> = MissedTaskV2<AccountOf<T>, TaskId<T>>;
```
Comment (justinzhou93) [inline pallets/automation-time/src/lib.rs:93]: should these belong in the types file as well?

File: pallets/automation-time/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+			self.execution_times.len() == other.execution_times.len() &&
+			self.execution_times.capacity() == other.execution_times.capacity() &&
+			self.execution_times.to_vec() == other.execution_times.to_vec()
+	}
+}
+
+impl<AccountId: Ord, Balance: Ord, CurrencyId: Ord, MaxExecutionTimes: Get<u32>> Eq
+	for Task<AccountId, Balance, CurrencyId, MaxExecutionTimes>
+{
+}
```
Comment (justinzhou93) [inline pallets/automation-time/src/types.rs:96]: supposed to be empty?

**Review by justinzhou93 — COMMENTED** [review]
looks good for the most part, couple of questions

File: pallets/automation-time/src/lib.rs (hunk 1 above)
Comment (rhuttman) [reply pallets/automation-time/src/lib.rs:93]: No. They are the types w/ the generics from the pallet

File: pallets/automation-time/src/types.rs (hunk 2 above)
Comment (rhuttman) [reply pallets/automation-time/src/types.rs:96]: ya. This just tells it to use the PartialEq impl. I need it for defining the traits of the generics needed for this

File: pallets/automation-time/src/tests.rs [hunk 3]
```diff
@@ -28,6 +28,11 @@ use sp_runtime::{
 	AccountId32,
 };
 
+type ActionOf<T> = Action<AccountId, Balance, <T as Config>::CurrencyId>;
+type TaskOf<T> =
+	Task<AccountId, Balance, <T as Config>::CurrencyId, <T as Config>::MaxExecutionTimes>;
+type MissedTaskV2Of = MissedTaskV2<AccountId, sp_core::H256>;
```
Comment (nvengal) [inline pallets/automation-time/src/tests.rs:34]:
```suggestion
use crate::{ActionOf, MissedTaskV2Of, TaskOf};
```
Need to update uses of MissedTaskV2Of to specify the generic arg for this to work `MissedTaskV2Of::<Test>`

File: pallets/automation-time/src/benchmarking.rs [hunk 4]
```diff
@@ -26,6 +26,15 @@ use sp_runtime::traits::{AccountIdConversion, Saturating};
...
 use crate::Pallet as AutomationTime;
 
+type TaskOf<T> = Task<
+	<T as frame_system::Config>::AccountId,
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance,
+	<T as Config>::CurrencyId,
+	<T as Config>::MaxExecutionTimes,
+>;
+type MissedTaskV2Of<T> =
+	MissedTaskV2<<T as frame_system::Config>::AccountId, <T as frame_system::Config>::Hash>;
```
Comment (nvengal) [inline pallets/automation-time/src/benchmarking.rs:36]:
```suggestion
use crate::{MissedTaskV2Of, TaskOf}
```

[the remark this answers is not in the document]
Comment (rhuttman) [reply pallets/automation-time/src/benchmarking.rs:36]: idk why I didn't think this would work...

**Review by justinzhou93 — APPROVED** [review]

**Review by nvengal — APPROVED** [review]

**Pull request merged**